File: clients/ruby/test/runtime_metrics_test.rb
require "minitest/autorun"
require_relative "../lib/microscope_client"

class RuntimeMetricsTest < Minitest::Test
  def test_sample_returns_expected_shape
    metrics = MicroscopeClient::RuntimeMetrics.sample

    assert_equal "ruby.runtime", metrics[:name]
    assert_equal "ruby", metrics[:language]
    assert_equal "threads", metrics[:unit]
    assert_kind_of Integer, metrics[:value]
    assert metrics[:value] >= 1
    assert_kind_of Integer, metrics[:gc_count]
    assert_kind_of Integer, metrics[:heap_live_slots]
  end
end

File: clients/typescript/test/runtimeMetrics.test.ts
import test from "node:test";
import assert from "node:assert/strict";

import { isNodeRuntime, sampleRuntimeMetrics } from "../src/runtimeMetrics.ts";

test("isNodeRuntime() is true under node", () => {
  assert.equal(isNodeRuntime(), true);
});

test("sampleRuntimeMetrics() returns the expected shape", () => {
  const metrics = sampleRuntimeMetrics();

  assert.equal(metrics.name, "node.runtime");
  assert.equal(metrics.language, "node");
  assert.equal(metrics.unit, "%");
  assert.equal(typeof metrics.value, "number");
  assert.ok(metrics.value >= 0 && metrics.value <= 100);
  assert.equal(typeof metrics.memory_mb, "number");
  assert.ok((metrics.memory_mb as number) > 0);
  assert.equal(typeof metrics.rss_mb, "number");
  assert.ok((metrics.rss_mb as number) > 0);
});
